third_party/openthread/CMakeLists.txt

```diff
@@ -62,21 +61,35 @@ set(OT_NAT64_TRANSLATOR ${OTBR_NAT64} CACHE STRING "enable NAT64 translator" FOR
 set(OT_NETDATA_PUBLISHER ON CACHE STRING "enable netdata publisher" FORCE)
 set(OT_NETDIAG_CLIENT ON CACHE STRING "enable Network Diagnostic client" FORCE)
 set(OT_PLATFORM "posix" CACHE STRING "use posix platform" FORCE)
-set(OT_PLATFORM_DNSSD ${OTBR_DNSSD_PLAT} CACHE STRING "enable platform DNSSD" FORCE)
 set(OT_PLATFORM_NETIF ON CACHE STRING "enable platform netif" FORCE)
 set(OT_PLATFORM_UDP ON CACHE STRING "enable platform UDP" FORCE)
 set(OT_SERVICE ON CACHE STRING "enable service" FORCE)
 set(OT_SLAAC ON CACHE STRING "enable SLAAC" FORCE)
 set(OT_SRP_CLIENT ON CACHE STRING "enable SRP client" FORCE)
-set(OT_SRP_ADV_PROXY ${OTBR_OT_SRP_ADV_PROXY} CACHE STRING "enable SRP Advertising Proxy" FORCE)
 set(OT_TARGET_OPENWRT ${OTBR_OPENWRT} CACHE STRING "target on OpenWRT" FORCE)
 set(OT_TCP OFF CACHE STRING "disable TCP")
 set(OT_TREL ${OTBR_TREL} CACHE STRING "enable TREL" FORCE)
 set(OT_UDP_FORWARD OFF CACHE STRING "disable udp forward" FORCE)
 set(OT_UPTIME ON CACHE STRING "enable uptime" FORCE)
 
-if (OTBR_DNSSD_PLAT)
-    set(OT_BORDER_AGENT_SERVICE_NAME ${OTBR_MESHCOP_SERVICE_INSTANCE_NAME} CACHE STRING "set the border agent service base name" FORCE)
+if (OTBR_MDNS_USE_OT_CORE)
+    set(OT_DNSSD_DISCOVERY_PROXY ON CACHE STRING "enable DNS-SD Discovery Proxy" FORCE)
+    set(OT_DNSSD_SERVER ON CACHE STRING "enable DNS-SD server support" FORCE)
+    set(OT_MDNS ON CACHE STRING "OpenThread mDNS" FORCE)
+    set(OT_SRP_ADV_PROXY ON CACHE STRING "enable SRP Advertising Proxy" FORCE)
+    set(OT_TREL_MANAGE_DNSSD ON CACHE STRING "TREL to manage DNSSD and peer discovery" FORCE)
```

Comment:
One combination to consider is `OTBR_DNSSD_DISCOVERY_PROXY` along with `OTBR_MDNS_USE_OT_CORE`.

I think we want `OTBR_DNSSD_DISCOVERY_PROXY` to be explicitly off when `OTBR_MDNS_USE_OT_CORE` is enabled. Otherwise, the OTBR code will enable its own discovery proxy function by registering `otDnssdQuerySetCallbacks()`. In this case, the OT `Dnssd::Server` will use these functions. Basically, if the platform indicates it wants to act as a discovery proxy by registering callbacks, it will take precedence over the native discovery proxy, even if the feature is enabled.

Perhaps we can add a check here that if `OTBR_DNSSD_DISCOVERY_PROXY` is enabled, an error message is emitted for an inconsistent configuration.